packages/core/src/node/messaging/messaging-contribution.ts

```diff
@@ -63,10 +63,10 @@ export class MessagingContribution implements BackendApplicationContribution, Me
         });
     }
 
-    forward(spec: string, callback: (params: MessagingService.PathParams, connection: IConnection) => void): void {
+    forward(spec: string, callback: (params: MessagingService.PathParams, connection: IConnection, parameters?: any) => void): void {
         return this.wsChannel(spec, (params, channel) => {
             const connection = launch.createWebSocketConnection(channel);
-            callback(params, connection);
+            callback(params, connection, channel.parameters);
```

Comment:
It looks strange. Now, there are `params` and `parameters`. What is the difference? Why `params` cannot be used?

Reply:
indeed, what about introducing a params type that extends the `MessagingService.PathParams`?